app/src/main/java/net/squanchy/service/repository/firebase/FirestoreEventRepository.kt

```diff
@@ -2,59 +2,61 @@ package net.squanchy.service.repository.firebase
 
 import io.reactivex.Completable
 import io.reactivex.Observable
-import io.reactivex.functions.BiFunction
+import io.reactivex.functions.Function3
 import io.reactivex.schedulers.Schedulers
 import net.squanchy.schedule.domain.view.Event
-import net.squanchy.service.firebase.FirebaseDbService
 import net.squanchy.service.firestore.FirestoreDbService
 import net.squanchy.service.firestore.model.conferenceinfo.FirestoreVenue
 import net.squanchy.service.firestore.model.schedule.FirestoreEvent
+import net.squanchy.service.firestore.model.schedule.FirestoreFavorite
 import net.squanchy.service.firestore.toEvent
 import net.squanchy.service.repository.EventRepository
 import net.squanchy.support.lang.Checksum
 import org.joda.time.DateTimeZone
 
-class FirebaseEventRepository(
-    private val dbService: FirebaseDbService,
+class FirestoreEventRepository(
     private val firestoreDbService: FirestoreDbService,
     private val checksum: Checksum
 ) : EventRepository {
 
     override fun event(eventId: String, userId: String): Observable<Event> {
         val eventObservable = firestoreDbService.event(eventId)
         val timeZoneObservable = firestoreDbService.venueInfo().map { it.extractTimeZone() }
+        val favoritesObservable = firestoreDbService.favorites(userId)
 
         return Observable.combineLatest(
             eventObservable,
             timeZoneObservable,
-            BiFunction(::combineIntoEvent)
+            favoritesObservable,
+            Function3(::combineIntoEvent)
         ).subscribeOn(Schedulers.io())
     }
 
-    private fun FirestoreVenue.extractTimeZone(): DateTimeZone {
-        return DateTimeZone.forID(timezone)
-    }
+    private fun FirestoreVenue.extractTimeZone() = DateTimeZone.forID(timezone)
+
+    private fun combineIntoEvent(event: FirestoreEvent, timeZone: DateTimeZone, favorites: List<FirestoreFavorite>): Event =
+        event.toEvent(checksum, timeZone, favorites.includes(event))
 
-    private fun combineIntoEvent(event: FirestoreEvent, timeZone: DateTimeZone): Event = event.toEvent(checksum, timeZone)
+    private fun combineIntoEvents(events: List<FirestoreEvent>, timeZone: DateTimeZone, favorites: List<FirestoreFavorite>): List<Event> =
```

Comment:
Probably my bad, but this should go after `override fun events()`, and `combineIntoEvent()` should go after this (top-down order)